src/screens/nano/NanoContractDetail.js

```diff
@@ -0,0 +1,202 @@
+        if (ignore) {
+          // This is to prevent setting a state after the componenet has been already cleaned
+          return;
+        }
+        setHistory(data.history);
+        setLoadingHistory(false);
+      } catch (e) {
+        if (ignore) {
+          // This is to prevent setting a state after the componenet has been already cleaned
+          return;
+        }
+        // Error in request
+        setErrorMessage('Error getting nano contract history.');
+        setLoadingHistory(false);
+      }
+    }
+    loadBlueprintInformation();
+    loadNCHistory();
+
+    return () => {
+      ignore = true;
+    }
+  }, [ncId]);
+
+
+  if (errorMessage) {
+    return <p className='text-danger mb-4'>{errorMessage}</p>;
+  }
+
+  if (loadingHistory || loadingDetail) {
+    return <Loading />;
+  }
+
+  const loadTable = () => {
+    return (
+      <div className="table-responsive mt-5">
+        <table className="table table-striped" id="tx-table">
+          <thead>
+            <tr>
+              <th className="d-none d-lg-table-cell">Hash</th>
+              <th className="d-none d-lg-table-cell">Timestamp</th>
+              <th className="d-table-cell d-lg-none" colSpan="2">Hash<br/>Timestamp</th>
+            </tr>
+          </thead>
+          <tbody>
+            {loadTableBody()}
+          </tbody>
+        </table>
+      </div>
+    );
+  }
+
+  const loadTableBody = () => {
+    return history.map((tx, idx) => {
+      // For some reason this API returns tx.hash instead of tx.tx_id like the others
+      tx.tx_id = tx.hash;
+      return (
+        <TxRow key={tx.tx_id} tx={tx} />
+      );
+    });
+  }
+
+  const renderNCAttributes = () => {
+    return (
+      <div className="table-responsive">
+        <table className="table table-striped table-bordered" id="attributes-table">
+          <thead>
+            <tr>
+              <th className="d-lg-table-cell">Name</th>
+              <th className="d-lg-table-cell">Value (or Type)</th>
+            </tr>
+          </thead>
+          <tbody>
+            {renderAttributes()}
+          </tbody>
+        </table>
+      </div>
+    );
+  }
+
+  const renderAttributeValue = (name, data) => {
+    // If the attribute is a dict, it won't return the value of it
+    // it will return an error message {errmsg: 'field not found'}
+    // In this case, we will show only the attribute type
+    // In the future, we plan to have a query feature, so the user can
+    // query these attributes until they get the value they need
+    return 'value' in data ? data.value : blueprintInformation.attributes[name];
+  }
+
+  const renderAttributes = () => {
+    return Object.entries(ncState.fields).map(([name, data]) => {
+      return (
+        <tr key={name}>
+          <td>{name}</td>
+          <td>{renderAttributeValue(name, data)}</td>
+        </tr>
+      );
+    });
+  }
+
+  // TODO identify that attribute is a token and show as NFT, in case it is.
```

Comment:
This TODO is for another PR?

Reply:
Yes. Right now we don't have the information when an attribute is a token. I discussed this with Marcelo and we will improve this on hathor-core, so we can better render information on the client.